app/controllers/users_controller.rb

```diff
@@ -1,6 +1,39 @@
 class UsersController < ApplicationController
+
+  before_action :authenticate_user!
+
+  # Only users and admins can edit user's profile
+  def edit
+  end
+
+  def show
+     @user = User.find(params[:id])
+  end
+
+  # Only users and admins can update user's profile
+  def update
+    if current_user.update_attributes(user_params)
+      redirect_to edit_user_registration_path
+  end
```

Comment:
finally got it...so sorry I went through that code a couple of times and could not find the extra 'end'.  Thanks for Ken's QA!  New version being updated and sent now.